docs/run-random-beacon.adoc

```diff
@@ -0,0 +1,402 @@
+:toc: macro
+
+= Run Random Beacon
+
+:icons: font
+:numbered:
+toc::[]
+
+== System Considerations
+
+- It is paramount that Keep nodes remain available to the Keep network. We encourage a stable and redundant internet connection.
+- A connection to a production grade self-hosted or third party Ethereum node deployment.
+- Persistent and redundant storage that will survive a VM or container rotation, and disk failure.
+- Each random beacon client running on the network requires a unique Ethereum operator account.
+- Each random beacon client running on the network requires a unique IP address or a unique
+  application port running under the same IP.
+- Machines:
+
+[%header,cols=2*]
+|===
+|Cloud Provider
+|Machine Type
+
+|Google Cloud
+|n1-standard-2
+
+|AWS
+|m5.large
+
+|Azure
+|D2s v3
+
+|Self-hosted
+|2 vCPU / 4 GiB RAM / 1 GiB Persistent Storage
+|===
+
+== Configuration
+
+=== Network
+
+Default port mappings.
+
+[%header,cols=2*]
+|===
+|Egress
+|Port
+
+|Ethereum Network
+|`8545` / `8546`
+
+|Keep Network
+|`3919`
+|===
+
+[%header,cols=2*]
+|===
+|Ingress
+|Port
+
+|Keep Network
+|`3919`
+|===
+
+If you set a different `port` in your keep-client configuration, or configure `peers` with
+non-default ports configured, firewall rules will need to be adjusted accordingly.
+
+=== Application
+
+Application configurations are stored in a `.toml` file and passed to the application run command
+ with the `--config` flag.
+
+==== Sample
+
+```
+# Ethereum host connection info.
+[ethereum]
+  URL = "ws://127.0.0.1:8546"
+  URLRPC = "http://127.0.0.1:8545"
+
+# Keep operator Ethereum account.
+[ethereum.account]
+  Address = "0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8AAAAAAAAA"
+  KeyFile = "/Users/someuser/ethereum/data/keystore/UTC--2018-03-11T01-37-33.202765887Z--AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8AAAAAAAAA"
+
+# Keep contract addresses configuration.
+[ethereum.ContractAddresses]
+  # Hex-encoded address of KeepRandomBeaconOperator contract
+  KeepRandomBeaconOperator = "0xBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB"
+  # Hex-encoded address of TokenStaking contract
+  TokenStaking = "0xCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+  # Hex-encoded address of KeepRandomBeaconService contract. Only needed
+  # in cases where the client's utility functions will be used (e.g., the
+  # relay subcommand).
+  KeepRandomBeaconService = "0xDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD"
+
+# Keep network configuration.
+[LibP2P]
+  Peers = ["/ip4/127.0.0.1/tcp/3919/ipfs/njOXcNpVTweO3fmX72OTgDX9lfb1AYiiq4BN6Da1tFy9nT3sRT2h1", "/dns4/some-keep-host.com/tcp/3919/ipfs/njOXcNpVTweO3fmX72OTgDX9lfb1AYiiq4BN6Da1tFy9nT3sRT2h1"]
+  Port  = 3920
+  # Override the node's default addresses announced in the network
+  AnnouncedAddresses = ["/dns4/example.com/tcp/3919", "/ip4/80.70.60.50/tcp/3919"]
+
+# Storage is encrypted
```

Comment:
Probably worth clarifying what we mean here. Are we saying the storage dir should be encrypted, or that the data written to storage is encrypted?

Reply:
I'm actually not sure what's happening here, lol.  Or of the logic used for `current` vs `archive`

Reply:
I asked Piotr for a one-liner somewhere up the PR.

Reply:
I think we can park this in his court via issue after merging, probably.

Short version though I believe:
- The contents of the storage dir are encrypted before they hit disk and decrypted on start.
- `current` is material for open keep/beacon groups, `archive` is material for closed/terminated keep/beacon groups.

Not 100% sure that's true off the top of my head though, so I think we leave it for now.